src/EventStore.Core/Services/Storage/ReaderIndex/EventFilter.cs

```diff
@@ -7,6 +7,8 @@
 namespace EventStore.Core.Services.Storage.ReaderIndex {
 	public static class EventFilter {
 		public static IEventFilter None => new AlwaysAllowStrategy();
+		public static IEventFilter DefaultAllFilter { get; } =
+			new NotStrategy(StreamName.Prefixes(SystemStreams.EpochInformationStream));
```

Comment:
@timothycoleman we can consider always allowing if the first character is '$' then fall back to the full prefix check. This is much faster for user streams which make up the majority of the transaction log. There's a similar implementation on `master`: https://github.com/EventStore/EventStore/blob/d5fe9fa1d9a72f0f9dcca4edb9efc90397c073bf/src/EventStore.Core/Services/Storage/ReaderIndex/EventFilter.cs#L66

Reply:
made a bespoke filter for the all default that just does the inequality check on the stream name, it does perform a lot better. there doesn't seem to be an additional boost from checking the $ prefix first, probably because here (unlike in master) we are just doing equality and not a prefix check